cirq/devices/device.py

```diff
@@ -12,20 +12,44 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-from typing import TYPE_CHECKING
-
 import abc
-
+from typing import TYPE_CHECKING, Optional, AbstractSet
 
 if TYPE_CHECKING:
     import cirq
 
-# Note: circuit/schedule types specified by name to avoid circular references.
-
 
 class Device(metaclass=abc.ABCMeta):
     """Hardware constraints for validating circuits."""
 
+    def qubit_set(self) -> Optional[AbstractSet['cirq.Qid']]:
+        """Returns a set or frozenset of qubits on the device, if possible.
+
+        This method returns None when the set of qubits is unavailable. For
+        example, `cirq.UnconstrainedDevice` returns `None` because allows any
+        and all qubits; it has no finite qubit set to return.
+
+        Returns:
+            If the device has a finite set of qubits, then a set or frozen set
+            of all qubits on the device is returned.
+
+            If the device has no well defined finite set of qubits (e.g.
+            `cirq.UnconstrainedDevice` has this property), then `None` is
+            returned.
+        """
+
+        # Compatibility hack to work with devices that were written before this
+        # method was defined.
```

Comment:
As @dstrain115 mentioned, it'd be nice to have a plan for where we are going with `Device` so that we can converge on an interface in the future and get rid of this hack. Maybe create an issue to track this and reference the issue number here.

Reply:
Acknowledged.